app/javascript/article-form/components/ImageUploader.jsx

```diff
@@ -123,6 +123,48 @@ export const ImageUploader = () => {
     });
   }
 
+  function handleNativeMessage(e) {
+    let message = {};
+    try {
+      message = JSON.parse(e.target.value);
+    } catch (e) {
+      console.log(e); // eslint-disable-line no-console
+      return;
+    }
+
+    switch (message.action) {
```

Comment:
Missing tests (blocking): It should be possible to add tests for this code by firing change events on the input it's bound to. Happy to help if you're not sure how to approach this.